server/handler.go

```diff
@@ -103,6 +106,30 @@ func (h *Handler) ComQuery(
 
 	var r *sqltypes.Result
 	var proccesedAtLeastOneBatch bool
+
+	rowchan := make(chan sql.Row)
+	errchan := make(chan error)
+
+	// This goroutine will be selected giving a change to Vitess to call the
+	// handler.CloseConnection callback and enforcing the Timeout if configured
+	go func(rowc chan sql.Row, errc chan error) {
+		for {
+			row, err := rows.Next()
+			if err != nil {
+				errc <- err
+				return
+			}
+			rowc <- row
+		}
+	}(rowchan, errchan)
```

Comment:
It isn't necessary to bind these to parameters—the values don't change during the lifetime of the closure.

```go
go func() {
   …
}()
```

Reply:
Done in next commit.